spec/lib/answer_composition/pipeline/bedrock_structured_answer_composer_spec.rb

```diff
@@ -0,0 +1,97 @@
+RSpec.describe AnswerComposition::Pipeline::BedrockStructuredAnswerComposer, :chunked_content_index do
+  describe ".call" do
+    let(:question) { build :question }
+    let(:context) { build(:answer_pipeline_context, question:) }
+    let(:response) do
+      [
+        {
+          output: {
+            message: {
+              role: "assistant",
+              content: [
+                { text: "VAT (Value Added Tax) is a tax applied to most goods and services in the UK." },
+              ],
+            },
+          },
+          stop_reason: "end_turn",
+          usage: {
+            input_tokens: 10,
+            output_tokens: 20,
+            total_tokens: 30,
+          },
+          metrics: {
+            latency_ms: 999,
+          },
+        },
+      ]
+    end
+
+    it "makes a successful request to Bedrock" do
+      request = stub_bedrock_request(:converse, response)
+      described_class.call(context)
+      expect(request[0][:http].status_code).to be 200
+    end
+
+    context "when a successful response is received" do
+      it "calls Bedrock converse endpoint and assigns the correct values to the context's answer" do
```

Comment:
```suggestion
      it "uses Bedrock converse endpoint to assign the correct values to the context's answer" do
```

Just a minor edit to reflect that Bedrock is used to populate the answer.

---

This test seems good. I think a decision for us is whether we want to just assert that given a particular question we get an answer back (as per the method I suggested for the test helper) or whether we want to assert any of the data that is pushed to bedrock (tool configuration for example)

I think it's ok that we don't assert the tool config - as it is config after all - but do you know if that being invalid would raise an exception? I'm kinda hoping there isn't a need to assert it because AWS validates it has the right object structure.

Reply:
Yeah, from everything I've seen so far, AWS always validates the structure of anything pushed to Bedrock, and returns an error if it isn't what's expected/required.